Storage/Rust/RustPlaces.swift

```diff
@@ -353,6 +353,12 @@ public class RustPlaces: BookmarksHandler, HistoryMetadataObserver {
         return deferred
     }
 
+    public func registerWithSyncManager() {
+        writerQueue.async {
+            self.api?.registerWithSyncManager()
```

Comment:
Just wanted so say I'm excited to finally see our sync manager here!! 🔥 🔥

Reply:
would be good to have `self` as weak or unowned. See above for which one to use